libs/execution_context/ebpf_maps.c

```diff
@@ -1179,6 +1185,199 @@ _update_stack_map_entry(
     return result;
 }
 
+static _Requires_lock_held_(ring_buffer_map->core_map.lock) void _ebpf_ring_buffer_map_signal_async_query_complete(
+    _In_ ebpf_core_ring_buffer_map_t* ring_buffer_map)
+{
+    EBPF_LOG_ENTRY();
+    // Skip if no async_contexts have ever been queued.
+    if (!ring_buffer_map->async_contexts_trip_wire) {
+        return;
+    }
+
+    ebpf_core_map_t* map = &ring_buffer_map->core_map;
```

Comment:
Can be simplified if we only permit a single ring buffer consumer.

Reply:
Will be addressed in #669